main/accounts/index.ts

```diff
@@ -521,9 +525,37 @@ export class Accounts extends EventEmitter {
     const request = currentAccount.requests[handlerId] && currentAccount.requests[handlerId].type === 'transaction'
     if (!request) return cb(new Error(`Could not locate request ${handlerId}`))
 
+    const lastSignerType = getSignerType(currentAccount.lastSignerType)
     const signer = currentAccount.getSigner()
-    if (!signer) return cb(new Error('No signer'))
+    
+    // hardware signers - handle locked and other states requiring user action
+    if (!signer && isHardwareSigner(lastSignerType)) {
+      const lockedSigners = Object.keys(store('main.signers'))
+        .map((s) => store('main.signers', s))
+        .filter(({ type, status }) => {
+          const isLocked = {
```

Comment:
I like the idea of just popping up the signer in any case that we're not ready to sign. would it just be simpler here to check if the signer is ready (currently means Status === OK)?

Reply:
also that way we can uniformly apply it to all signers (not just hardware) and basically turn it into a one liner